### PR TITLE
Update agents listing service with "CanReAttest" filter

### PR DESCRIPTION
**Pull request by guilhermocc**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
spire-server agents listing.

**Description of change**
- Add "CanReAttest" boolean filter in list agents method
- Update spire-server `agent` and `entry` commands to add new response fields: registration entry `hint` and agent `canReAttest`.

**Which issue this PR fixes**
Ongoing work for #1836

**Dependencies**
⚠️ This PR depends on https://github.com/spiffe/spire-api-sdk/pull/36. After it's merge, go mod file should be updated to remove temporary local spire-api-sdk reference. ⚠️

### CODE REVIEW

**Comment by faisal-memon** [comment]
I can review

File: cmd/spire-server/cli/agent/agent_test.go [hunk 1]
```diff
@@ -266,7 +266,7 @@ func TestList(t *testing.T) {
 			expectedReturnCode:   0,
```
Comment (MarcosDY) [inline cmd/spire-server/cli/agent/agent_test.go:266]: can you add a test case where an agent that is re_attestable is returned?

File: cmd/spire-server/cli/entry/create_test.go [hunk 2]
```diff
@@ -289,6 +289,7 @@ StoreSvid        : true
       },
       "entry": {
         "id": "entry-id",
+        "hint": "",
```
Comment (MarcosDY) [inline cmd/spire-server/cli/entry/create_test.go:292]: why hint is required here? looks like it is not related with this PR (the same comment is for all files with the same change)

Comment (guilhermocc) [reply cmd/spire-server/cli/entry/create_test.go:292]: The new version of spire-api-sdk comes with the hint field change; that's why I had to include those here. This field will be effectively included in https://github.com/spiffe/spire/pull/3926

File: cmd/spire-server/cli/agent/agent_test.go (hunk 1 above)
Comment (guilhermocc) [reply cmd/spire-server/cli/agent/agent_test.go:266]: 0db980d

**Review by MarcosDY — APPROVED** [review]

**Pull request merged**